source/transactions/tests/pin-mongos.yml

```diff
@@ -0,0 +1,298 @@
+# Test that all the operations go to the same mongos.
+#
+# The assertion is implicit: that all the read operations succeed.
+# If the driver does not properly pin to a single mongos then one of the
+# operations in a transaction will eventually be sent to a different mongos,
+# which is unaware of the transaction, and the mongos will return a command
+# error. An example of such an error is:
+# {
+#   'ok': 0.0,
+#   'errmsg': 'cannot continue txnId -1 for session 28938f50-9d29-4ca5-8de5-ddaf261267c4 - 47DEQpj8HBSa+/TImW+5JCeuQeRkm5NMpJWZG3hSuFU= with txnId 1',
+#   'code': 251,
+#   'codeName': 'NoSuchTransaction',
+#   'errorLabels': ['TransientTransactionError']
+# }
```

Comment:
In this example, the "different" mongos has seen a higher transaction (txnId 1) than what's in the request (txnId -1), which I wouldn't expect to happen normally.

If you want, you could give an example of the typical case, where the request's transaction number is higher than what the "different" mongos has seen (which is why I'd expect to see the 'missing startTransaction' error).

Reply:
This is the error I get back from the server when attempting to continue a transaction on a different Mongos. I think the problem here is just that the server's error message is unclear.  txnId -1  is not txnNumber that the driver sent to the server, it's the server's txnId value for this session.

Reply:
Hm, you're right, the source code on mongos is a bit odd. I'll file a server ticket for it.

Reply:
Filed https://jira.mongodb.org/browse/SERVER-38420.

Reply:
Thanks!